src/dapps/saga.ts

```diff
@@ -26,10 +48,64 @@ export function* handleOpenDapp(action: PayloadAction<DappSelectedAction>) {
   }
 }
 
+export function* handleFetchDappsList() {
+  const dappsListApiUrl = yield select(dappsListApiUrlSelector)
+  const address = yield select(walletAddressSelector)
+  const language = yield select(currentLanguageSelector)
+  const shortLanguage = language.split('-')[0]
+
+  const response = yield call(
+    fetch,
+    `${dappsListApiUrl}?language=${shortLanguage}&address=${address}`,
+    {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    }
+  )
+
+  if (response.ok) {
+    try {
+      const result: {
+        applications: Application[]
+        categories: DappCategory[]
+        featured: Application
+      } = yield call([response, 'json'])
+
+      const dappsList: Dapp[] = result.applications.map((application) => {
+        return {
+          id: application.id,
+          categoryId: application.categoryId,
+          name: application.name,
+          iconUrl: application.logoUrl,
+          description: application.description,
+          dappUrl: application.url.replace('{{address}}', address ?? ''),
+          isFeatured: application.id === result.featured.id,
+        }
+      })
+
+      yield put(fetchDappsListCompleted({ dapps: dappsList, categories: result.categories }))
+    } catch (error) {
+      Logger.error(TAG, 'Could not parse dapps response', error)
+      yield put(fetchDappsListFailed({ error: 'Could not parse dapps' }))
+    }
+  } else {
+    yield put(fetchDappsListFailed({ error: 'Could not fetch dapps' }))
+  }
+}
+
 export function* watchDappSelected() {
   yield takeLatest(dappSelected.type, handleOpenDapp)
 }
 
+export function* watchFetchDappsList() {
+  yield takeLeading(fetchDappsList.type, handleFetchDappsList)
+}
+
 export function* dappsSaga() {
   yield spawn(watchDappSelected)
+  yield spawn(watchFetchDappsList)
+  yield put(fetchDappsList())
```

Comment:
Nothing to do now, but there's an edge case if this fetch fails, some things relying on the data to be present won't work as expected.
That's one downside of doing the manual fetching instead of having a Redux Toolkit Query style hook which we would use where we need the data and would re-trigger it as needed.

Reply:
yeah true, if this doesn't load correctly on app start + the user does not visit the dapps page (or for some reason it fails there too) then we will always show the dapps disclaimer :/ i wonder if we should only show the "not in dapps list" disclaimer if there are dapps in the dapplist?

Reply:
i think i'll leave this - we're not clearing the dappsList on refetch, so this edge case would be limited to users who have never been able to fetch the dapps list, and in that case i think it's technically correct to keep the disclaimer if the dapplist is empty